File: binarySearch.js
function binario(number, n) {
    let max = number.length;
    let min = 0;
    let mid = 0;
    
    for (let i = 1; i < Math.floor(number.length / 2); i++) {
        mid = Math.floor((max+min)/2)
        if (number[mid] == n) {
            console.log(`The program ran ${i} and the item ${n} is at ${mid}`);
            break;
        } else if (n < number[mid]) {
            max = mid;
            mid = Math.floor((min + max) / 2);
            
        } else if (n > number[mid]) {
            min = mid;
            mid = Math.floor((min + max) / 2);
        };
    };
};

binario([1,2,3,4,5,6,7,8,9,10], 5);

File: Python/binarySearch.py
import math

def binarySearch(arr, num):
    min = 0
    max = len(arr)
    mid = 0
    i = 1
    
    while i < math.floor(len(arr)/2):
        mid = math.floor((min + max) / 2)
        if arr[mid] == num:
            return(f"The loop ran {i} times and the number {num} is at the index {mid}")
        elif arr[mid] > num:
            max = mid-1
        elif arr[mid] < num:
            min = mid+1
        i=i+1

    return "Coudn't find the given number in the array..."

print(binarySearch([1,2,3,4,5,6,7,8,9,10], 1))